cluster-autoscaler/processors/status/scale_up_status_processor.go

```diff
@@ -100,6 +100,43 @@ func (p *NoOpScaleUpStatusProcessor) Process(context *context.AutoscalingContext
 func (p *NoOpScaleUpStatusProcessor) CleanUp() {
 }
 
+// CombinedScaleUpStatusProcessor is a list of ScaleUpStatusProcessor
+type CombinedScaleUpStatusProcessor struct {
+	processors []ScaleUpStatusProcessor
+}
+
+// NewCombinedScaleUpStatusProcessor construct CombinedScaleUpStatusProcessor.
+func NewCombinedScaleUpStatusProcessor(processors []ScaleUpStatusProcessor) *CombinedScaleUpStatusProcessor {
+	var scaleUpProcessors []ScaleUpStatusProcessor
+	for _, processor := range processors {
+		if processor != nil {
+			scaleUpProcessors = append(scaleUpProcessors, processor)
+		}
+	}
+	return &CombinedScaleUpStatusProcessor{scaleUpProcessors}
+}
+
+// AddProcessor append processor to the list.
+func (p *CombinedScaleUpStatusProcessor) AddProcessor(processor ScaleUpStatusProcessor) {
+	if processor != nil {
```

Comment:
When you check nil for an interface, you have to do this:
processor != nil && !reflect.ValueOf(processor).IsNil()

Reply:
In fact, doing it using the suggestion might also break if the interface value is a struct and not a pointer.
I think this is better discussed in bigger scale, so no action required here.

[Let's leave the discussion to the other comment since they are the same]